File: prog/osloware/mama/src/libs/util/caskyn.f

C======================================================================

      SUBROUTINE CASKYN(MESAG,*)

C         mesag:    question to be asked (character string)
C         return 1: answer = N/n/0/<return>
C         return:   answer = Y/y/1

      CHARACTER*(*) MESAG
      CHARACTER*40  ANS
      INTEGER       NCA


10    CALL CASK2(MESAG,ANS,NCA,1)
      IF (NCA.EQ.0
     +   .OR.ANS(1:1).EQ.'N'
     +   .OR.ANS(1:1).EQ.'n'
     +   .OR.ANS(1:1).EQ.'0') RETURN 1
      IF (ANS(1:1).EQ.'Y'
     +   .OR.ANS(1:1).EQ.'y'
     +   .OR.ANS(1:1).EQ.'1') RETURN
      GO TO 10
      END

File: mama/src/minig_x/caskyn.c
#include <stdio.h>
#include "minig_x/util.h"

/* ====================================================================== */
int caskyn(char *mesag)
{
  /* mesag:     question to be asked (character string)
     returns 1: answer = Y/y/1   0: answer = N/n/0/<return> */

  char ans[16];
  int iii=1;
  while (cask(mesag, ans, &iii)) {
    if (ans[0] == 'N' || ans[0] == 'n' || ans[0] == '0') return 0;
    if (ans[0] == 'Y' || ans[0] == 'y' || ans[0] == '1') return 1;
  }
  return 0;
} /* caskyn */
